diffsims/crystallography/_diffracting_vector.py

```diff
@@ -0,0 +1,181 @@
+        This, ``xyz``, or ``hkl`` is required.
+    intensity : numpy.ndarray, list, or tuple, optional
+        Intensity of the diffraction vector(s). Default is ``None``.
+    rotation : orix.quaternion.Rotation, optional
+        Rotation matrix previously applied to the reciprocal lattice vector(s) and the
+        lattice of the phase. Default is ``None`` which corresponds to the
+        identity matrix.
+
+
+    Examples
+    --------
+    >>> from diffpy.structure import Atom, Lattice, Structure
+    >>> from orix.crystal_map import Phase
+    >>> from diffsims.crystallography import DiffractingVector
+    >>> phase = Phase(
+    ...     "al",
+    ...     space_group=225,
+    ...     structure=Structure(
+    ...         lattice=Lattice(4.04, 4.04, 4.04, 90, 90, 90),
+    ...         atoms=[Atom("Al", [0, 0, 1])],
+    ...     ),
+    ... )
+    >>> rlv = DiffractingVector(phase, hkl=[[1, 1, 1], [2, 0, 0]])
+    >>> rlv
+    ReciprocalLatticeVector (2,), al (m-3m)
+    [[1. 1. 1.]
+     [2. 0. 0.]]
+
+    """
+
+    def __init__(self, phase, xyz=None, hkl=None, hkil=None, intensity=None):
+        super().__init__(phase, xyz=xyz, hkl=hkl, hkil=hkil)
+        if intensity is None:
+            self._intensity = np.full(self.shape, np.nan)
+        elif len(intensity) != self.size:
+            raise ValueError("Length of intensity array must match number of vectors")
+        else:
+            self._intensity = np.array(intensity)
+
+    def __getitem__(self, key):
+        new_data = self.data[key]
+        dv_new = self.__class__(self.phase, xyz=new_data)
+
+        if np.isnan(self.structure_factor).all():
+            dv_new._structure_factor = np.full(dv_new.shape, np.nan, dtype="complex128")
+
+        else:
+            dv_new._structure_factor = self.structure_factor[key]
+        if np.isnan(self.theta).all():
+            dv_new._theta = np.full(dv_new.shape, np.nan)
+        else:
+            dv_new._theta = self.theta[key]
+        if np.isnan(self.intensity).all():
+            dv_new._intensity = np.full(dv_new.shape, np.nan)
+        else:
+            slic = self.intensity[key]
+            if not hasattr(slic, "__len__"):
+                slic = np.array(
+                    [
+                        slic,
+                    ]
+                )
+            dv_new._intensity = slic
+
+        return dv_new
+
+    @property
+    def basis_rotation(self):
+        """
+        Returns the lattice basis rotation.
+        """
+        return Rotation.from_matrix(self.phase.structure.lattice.baserot)
+
+    def rotate_with_basis(self, rotation):
+        """Rotate both vectors and the basis with a given `Rotation`.
+        This differs from simply multiplying with a `Rotation`,
+        as that would NOT update the basis.
+
+        Parameters
+        ----------
+        rot : orix.quaternion.Rotation
+            A rotation to apply to vectors and the basis.
+        """
```

Comment:
I would suggest, and highly appreciate, and explanation that rotating the lattice may violate orix' assumption of crystal axes alignment (e.g. in a *Notes* section). This may lead to undefined behavior when applying the symmetry operations of the phase in the returned vectors.

Also, a description of the returned vectors in a dedicated section would be preferably.

Reply:
Not for end-users, but for the maintainers of diffsims.